### PR TITLE
Bugfix/1275

### PR DESCRIPTION
**Pull request by SergeyG22**
Place the total number of player colors should be the current number of players. The error was caused by an out-of-bounds vector.

### CODE REVIEW

File: client/lobby/CSelectionBase.cpp [hunk 1]
```diff
@@ -388,23 +388,21 @@ CFlagBox::CFlagBoxTooltipBox::CFlagBoxTooltipBox(std::shared_ptr<CAnimation> ico
...
-
 	labelTeamAlignment = std::make_shared<CLabel>(128, 30, FONT_MEDIUM, ETextAlignment::CENTER, Colors::YELLOW, CGI->generaltexth->allTexts[657]);
 	labelGroupTeams = std::make_shared<CLabelGroup>(FONT_SMALL, ETextAlignment::CENTER, Colors::WHITE);
 	for(int i = 0; i < SEL->getMapInfo()->mapHeader->howManyTeams; i++)
 	{
 		std::vector<ui8> flags;
 		labelGroupTeams->add(128, 65 + 50 * i, boost::str(boost::format(CGI->generaltexth->allTexts[656]) % (i+1)));
-
-		for(int j = 0; j < PlayerColor::PLAYER_LIMIT_I; j++)
+
```
Comment (kambala-decapitator) [inline client/lobby/CSelectionBase.cpp:397]: please avoid extra space on empty lines

```suggestion

```

**Comment by dydzio0614** [comment]
Thanks for contribution

**Review by Nordsoft91 — APPROVED** [review]

**Comment by kambala-decapitator** [comment]
made runtime test, noticed the following:
1. the original issue is not reproducible in develop branch, only in 1.1 release
2. with the applied fix not all team flags are displayed:

Screenshot 2022-12-27 at 10 58 27

and this is how it should be (and what I see in develop):
Screenshot 2022-12-27 at 10 59 27

I suggest to close the PR.

**Comment by kambala-decapitator** [comment]
I've also closed the original issue as I can't reproduce it

File: client/lobby/CSelectionBase.cpp [hunk 2]
```diff
@@ -388,23 +388,20 @@ CFlagBox::CFlagBoxTooltipBox::CFlagBoxTooltipBox(std::shared_ptr<CAnimation> ico
...
-
 	labelTeamAlignment = std::make_shared<CLabel>(128, 30, FONT_MEDIUM, ETextAlignment::CENTER, Colors::YELLOW, CGI->generaltexth->allTexts[657]);
 	labelGroupTeams = std::make_shared<CLabelGroup>(FONT_SMALL, ETextAlignment::CENTER, Colors::WHITE);
 	for(int i = 0; i < SEL->getMapInfo()->mapHeader->howManyTeams; i++)
 	{
 		std::vector<ui8> flags;
 		labelGroupTeams->add(128, 65 + 50 * i, boost::str(boost::format(CGI->generaltexth->allTexts[656]) % (i+1)));
-
-		for(int j = 0; j < PlayerColor::PLAYER_LIMIT_I; j++)
+		for(int j = 0; j < (int)SEL->getMapInfo()->mapHeader->howManyTeams; j++)
```
Comment (kambala-decapitator) [inline client/lobby/CSelectionBase.cpp:397]: - outer loop already uses the same condition
- `j` variable is used in `getPlayerInfo` call while `i` is for `TeamID`

this change doesn't look correct

**Comment by SergeyG22** [comment]
Thanks for the detailed comments, I'm just new and I didn't have enough time to sort out this problem in detail. I will try to study this problem well and if possible I will make a new pull request. I'm closing this pull request.

**Comment by kambala-decapitator** [comment]
no problem! All contributions are welcomed, even if they're not merged in the end :) Thank you!

P.S. I think it crashed due to the same reason as https://github.com/vcmi/vcmi/issues/1307#issuecomment-1366042645

**Comment by Nordsoft91** [comment]
@SergeyG22 , it may be quite dissaponting, but the thing is vcmi is complex and changing one place can easily brake something else. However, there are some tasks easy to start with, I believe (because changes needed are in quite isolated components). Map editor, for instance. Or adding new configurable objects. Let me know if you need help to get where to start from

**Comment by kambala-decapitator** [comment]
btw you can also reach us on slack to discuss something / ask questions

**Pull request closed without merging**